conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/ethereum/EthereumOperations.scala

```diff
@@ -0,0 +1,115 @@
+package tech.cryptonomic.conseil.indexer.ethereum
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import slickeffect.Transactor
+
+import tech.cryptonomic.conseil.common.config.Platforms.EthereumBatchFetchConfiguration
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+import tech.cryptonomic.conseil.common.ethereum.EthereumPersistence
+import tech.cryptonomic.conseil.common.ethereum.rpc.EthereumClient
+import tech.cryptonomic.conseil.indexer.config.{Custom, Depth, Everything, Newest}
+
+/**
+  * Ethereum operations for Lorre.
+  *
+  * @param ethereumClient Ethereum client instance
+  * @param persistence Ethereum persistence instance
+  * @param tx [[slickeffect.Transactor]] to perform a Slick operations on the database
+  * @param batchConf Batch processing configuration
+  */
+class EthereumOperations[F[_]: Concurrent](
+    ethereumClient: EthereumClient[F],
+    persistence: EthereumPersistence[F],
+    tx: Transactor[F],
+    batchConf: EthereumBatchFetchConfiguration
+) extends LazyLogging {
+
+  /**
+    * Start Lorre with mode defined with [[Depth]].
+    *
+    * @param depth Can be: Newest, Everything or Custom
+    */
+  def loadBlocksAndLogs(depth: Depth): Stream[F, Unit] =
+    Stream
+      .eval(tx.transact(persistence.getLatestIndexedBlock))
+      .zip(ethereumClient.getMostRecentBlockNumber.map(Integer.decode))
+      .flatMap {
+        case (block, mostRecentBlockNumber) =>
```

Comment:
Possibly `block` is too generic. Could we use something like `recentIndexedBlock` or similar descriptive name?
Is that useful?